odoo/addons/base/models/ir_actions_report.py

```diff
@@ -1121,6 +756,30 @@ def _get_rendering_context(self, report, docids, data):
         data['is_html_empty'] = is_html_empty
         return data
 
+    def _run_pdf_engine(
+        self, engine_name: str, html: str, report_ref: typing.Union[str, bool] = False,
+        landscape: bool = False, **kwargs) -> tuple[bytes, list[int]]:
+        """This function run associated pdf engine if exists with the given
+        html.
+
+        :param engine_name: The name of the pdf engine to use.
+        :param html: The html to convert to pdf.
+        :param report_ref: The report reference (opt.).
+        :param landscape: If the pdf should be in landscape mode (opt.).
+        :param kwargs: Additional arguments to pass to the pdf engine. (e.g. page size, margins, etc.)
+        :return: a tuple of (pdf, html_ids) where pdf is the generated pdf content and html ids to separate the
+            pdf content.
+        """
+        pdf_rendering_method = getattr(self, engine_name, None)
+        if pdf_rendering_method:
+            try:
+                pdf, html_ids = pdf_rendering_method(html, report_ref=report_ref, landscape=landscape, **kwargs)
+            except TypeError:
```

Comment:
Don't rely on type error and exceptions, just use `if callable(pdf_rendering_method)`.